src/sentry/api/endpoints/api_token_details.py

```diff
@@ -0,0 +1,92 @@
+from django.utils.decorators import method_decorator
+from django.views.decorators.cache import never_cache
+from rest_framework import serializers
+from rest_framework.fields import CharField
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import analytics
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import Endpoint, control_silo_endpoint
+from sentry.api.endpoints.api_tokens import ApiTokensEndpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.models.apitoken import ApiToken
+
+ALLOWED_FIELDS = ["name", "tokenId"]
+
+
+class ApiTokenNameSerializer(serializers.Serializer):
+    name = CharField(max_length=255, allow_blank=True, required=True)
+
+
+@control_silo_endpoint
+class ApiTokenDetailsEndpoint(Endpoint):
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+        "PUT": ApiPublishStatus.PRIVATE,
+        "DELETE": ApiPublishStatus.PRIVATE,
+    }
+    owner = ApiOwner.SECURITY
+    permission_classes = (IsAuthenticated,)
+
+    @method_decorator(never_cache)
+    def get(self, request: Request, token_id: int) -> Response:
+
+        user_id = ApiTokensEndpoint.get_appropriate_user_id(request=request)
```

Comment:
Having endpoints use each other for methods is not a pattern we frequently use. Instead you could promote this method to a module level function and have both endpoints use it that way.

Reply:
@nhsiehgit mentioned creating another ticket to move this into a separate file and group `api_token` files under a separate `endpoints/api_tokens/...` folder

Reply:
@ameliahsu have/can you create a jira task to do this and add it to your sprint?

Reply:
Oh wai t- i see some updated PRs referencing this ticket 🙏 
perfect, thank you!